### PR TITLE
chore(deps): Typescript 4.2.4

### PR DESCRIPTION
**Pull request by lukelooker**

### CODE REVIEW

File: packages/components/src/CopyToClipboard/CopyToClipboard.tsx [hunk 1]
```diff
@@ -75,21 +81,20 @@ export const CopyToClipboard: FC<CopyToClipboardProps> = (props) => {
...
-      >
-        {children}
-      </ButtonOutline>
-    ) : (
-      cloneElement(children, {
-        onClick: clickCopyButton,
-        ref: buttonRef,
-      })
-    )
+  const copyButton = isValidElement(children) ? (
```
Comment (lukelooker) [inline packages/components/src/CopyToClipboard/CopyToClipboard.tsx:84]: TS got smarter and discovers that previous type-guard of `typeof children === 'string'` is insufficient. Using `isValidElement` is a better solution so I inverted logic to leverage that and put it up-front.

File: packages/components/src/utils/getCurrentNode.ts [hunk 2]
```diff
@@ -36,7 +36,6 @@ export function getCurrentNode<E extends HTMLElement = HTMLElement>(
   elementOrRefObject: ElementOrRef<E>
 ): E | null {
   if (!elementOrRefObject) return null
-  // @ts-expect-error: Need to find a better way to do union below
```
Comment (lukelooker) [inline packages/components/src/utils/getCurrentNode.ts:39]: TS 4.2.x can now properly infer this union. 🎉

Comment (mdodgelooker) [reply packages/components/src/utils/getCurrentNode.ts:39]: Oh neat!

**Review by mdodgelooker — APPROVED** [review]
👍

**Pull request merged**